src/xb-database-manager.c

```diff
@@ -366,20 +439,23 @@ xb_database_manager_create_db_internal (XbDatabaseManager *self,
   XapianQueryParser *query_parser;
   DatabasePayload *payload;
   GFileMonitor *monitor;
+  const char *path;
 
-  g_assert (!g_hash_table_contains (priv->databases, xbdb.path));
+  path = xb_database_path (xbdb);
+
+  g_assert (!g_hash_table_contains (priv->databases, path));
+
+  if (xbdb.manifest_path)
```

Comment:
Also kinda awkward how this is kinda duplicating the logic in `xb_database_path`, but maybe not relevant if we can force all new shards to have manifest.